src/utils/generalFunctions.js

```diff
@@ -67,14 +68,93 @@ export function aboutPopup(call, classes = {}) {
   });
 }
 
+/* breaks a tool id into the alphabetical and numerical components */
+function breakToolId(tabId) {
+  const index = tabId.search(/\d/);
+  const alpha = tabId.substring(0, index - 1);
+
+  if (index < 0)
+    return [alpha];
+
+  const number = new Number(tabId.substring(index));
+
+  return [alpha, number];
+}
+
+/* get the structure from the tabsMap that will allow us
+ * to efficently organize tool ids.
+ */
+function computeIds(tabsMap) {
+  const res = {};
+
+  /* get a map of this format:
+   * {
+   *   Topics: { last: 3, busy: [2] },
+   *   VarsDebug: { last 2, busy: [1] },
+   * }
+   */
+  for (const [tabId] of tabsMap) {
+    const [alpha, number] = breakToolId(tabId);
```

Comment:
```suggestion
    const [toolName, tabIncrement] = breakToolId(tabId);
```

Reply:
Also, if you get rid of the `breakToolId` function, you can use something like:

```
const {scope, tabIncrement} = tabData;
```

Reply:
I don't think there's a strong reason to modify what we're storing there.